vehicle/polestar/identity.go

```diff
@@ -137,3 +146,47 @@ func (v *Identity) RefreshToken(token *oauth2.Token) (*oauth2.Token, error) {
 
 	return v.login()
 }
+
+func (v *Identity) confirmConsentAndGetCode(uri, resume string, param request.InterceptResult) (string, error) {
+
+	fmt.Printf("Attempting to retrieve authorization code again after user consent confirmation")
+
+	// Attempt to extract the "uid" parameter
+	v.Client.CheckRedirect, param = request.InterceptRedirect("uid", true)
+	defer func() { v.Client.CheckRedirect = nil }()
+
+	// Extract the user ID (UID) from the redirect parameters
+	var uid string
+	if uid, err := param(); err != nil || uid == "" {
```

Comment:
This uid is NOT the one from the line above. Due to := you‘re „shadowing“ it. Use = instead!

Reply:
Wow, I didn't expect that! Thanks for the tip. Learned something new :)
I'm curious to see what @loebse thinks about it.

Reply:
It is not enough to solve the problem of not reading the `uid`

The code proposal by @scyomantion works for me though. Are you making the changes accordingly @rostbeule?